sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -4136,6 +4136,29 @@ class Dataset[T] private[sql](
     new MergeIntoWriter[T](table, this, condition)
   }
 
+  /**
+   * Update rows in a table that match a condition.
+   *
+   * Scala Example:
+   * {{{
+   *   spark.table("source").update(Map("salary" -> lit(200)))
+   *    .where($"salary" === 100)
+   *    .execute()
+   *
+   * }}}
+   * @param assignments A Map of column names to Column expressions representing the updates
+   *     to be applied.
+   * @since 4.0.0
+   */
+  def update(assignments: Map[String, Column]): UpdateWriter[T] = {
```

Comment:
we can make the API even simpler:
```
def update(assignments: Map[String, Column]): Unit

def update(assignments: Map[String, Column], condition: Column): Unit

```

People can perform UPDATE with a single API call.

Reply:
@cloud-fan that also works, but what about possibility to add write options ?  Originally that was my thought to add a dedicated writer.

Reply:
well, we can add more overloads to include the `options`. My worry is many people are used to the common DataFrame pattern: `sourceDF.transformAPIs.write.writeConfigurationAPIs.insertInto/save/saveAsTable`. I think it's better to use a different API style for UPDATE and DELETE, as they don't have a source query.

Reply:
ok sounds good, thanks for the reply, ill make the change